### PR TITLE
Add docs for new `bot` resource

### PR DESCRIPTION
**Pull request by strideynet**
Dependent on https://github.com/gravitational/teleport/pull/35441

### CODE REVIEW

File: docs/pages/includes/machine-id/bot-spec.mdx [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+version: v1
+metadata:
+  # name is a unique identifier for the bot in the cluster.
+  name: robot
+spec:
+  # roles is a list of roles that the bot should be able to generate credentials
+  # for.
+  roles:
+  - editor
+  # traits controls the traits applied to the Bot user. These are
```
Comment (zmb3) [inline docs/pages/includes/machine-id/bot-spec.mdx:12]: These are what?

Comment (strideynet) [reply docs/pages/includes/machine-id/bot-spec.mdx:12]: Oops - forgot to commit the rest of this chunk apparently.

File: docs/pages/machine-id/reference/configuration.mdx [hunk 2]
```diff
@@ -388,3 +388,13 @@ type: kubernetes_secret
 # This must be in the same namespace that `tbot` is running in.
 name: my-secret
 ```
+
+## Bot Resource
```
Comment (ptgott) [inline docs/pages/machine-id/reference/configuration.mdx:392]:
```suggestion
## Bot resource
```

**Review by ptgott — APPROVED** [review]

File: docs/pages/includes/machine-id/bot-spec.mdx (hunk 1 above)
Comment (strideynet) [reply docs/pages/includes/machine-id/bot-spec.mdx]: c5db7ba04b

**Review by zmb3 — APPROVED** [review]

**Pull request merged**